soccer/WindowEvaluator.h

```diff
@@ -0,0 +1,75 @@
+//
+// Created by matt on 7/8/15.
+//
+
+#ifndef GT_ROBOJACKETS_ROBOCUP_WINDOWEVALUATOR_H
+#define GT_ROBOJACKETS_ROBOCUP_WINDOWEVALUATOR_H
+
+#include <boost/optional.hpp>
+#include <Geometry2d/Segment.hpp>
+#include <Geometry2d/Point.hpp>
+#include "Robot.hpp"
+#include "SystemState.hpp"
+
+class Window {
+public:
+  Window();
+
+  Window(double t0, double t1);
+
+  double a0;
+  double a1;
+  double t0;
+  double t1;
+  Geometry2d::Segment segment;
+  double shot_success;
+
+  bool operator == (const Window& other) const {
+    return a0 == other.a0
+        && a1 == other.a1
+        && t0 == other.t0
+        && t1 == other.t1;
+  }
+};
+
+using WindowingResult = std::pair<std::vector<Window>, boost::optional<Window>>;
+
+class WindowEvaluator {
+public:
+
+  WindowEvaluator(SystemState *systemState);
+
+  WindowingResult eval_pt_to_pt(Geometry2d::Point origin, Geometry2d::Point target, float targetWidth);
+
+  WindowingResult eval_pt_to_robot(Geometry2d::Point origin, Geometry2d::Point target);
+
+  WindowingResult eval_pt_to_opp_goal(Geometry2d::Point origin);
+
+  WindowingResult eval_pt_to_our_goal(Geometry2d::Point origin);
+
+  void obstacle_range(std::vector<Window>& windows, double& t0, double& t1);
+
+  void obstacle_robot(std::vector<Window>& windows, Geometry2d::Point origin, Geometry2d::Segment target, Geometry2d::Point bot_pos);
```

Comment:
I'm not opposed to replacing Geometry2d at some point, but it's honestly not that bad.  It does everything we want and aside from a couple bugs we've fixed, it works pretty well.